### PR TITLE
feat: exam

### PR DESCRIPTION
**Pull request by Michellah**

### CODE REVIEW

File: src/operations/exams/ExamCreate.tsx [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import {useMemo, useState} from "react";
+import {
+  NumberInput,
+  required,
+  SimpleForm,
+  TextInput,
+  useGetList,
+} from "react-admin";
+import {Create} from "@/operations/common/components";
+import authProvider from "@/providers/authProvider";
```
Comment (NyAndoMayah) [inline src/operations/exams/ExamCreate.tsx:10]: order your imports

File: src/operations/exams/ExamCreate.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import {DateTimeField} from "@/operations/common/components/fields";
+import {AutocompleteInput} from "@/ui/components/inputs";
+
+export const ExamCreate = () => {
+  const user = authProvider.getCachedWhoami();
+  const userRole = user.role;
```
Comment (NyAndoMayah) [inline src/operations/exams/ExamCreate.tsx]: object destructuration

File: src/operations/exams/ExamCreate.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    if (userRole !== "teacher") {
+      setSelectedTeacherId(value);
+    }
+  };
+
+  return (
+    <Create resource="exams" title="Création d'un examen">
+      <SimpleForm>
+        {userRole == "MANAGER" ? (
+          <>
```
Comment (NyAndoMayah) [inline src/operations/exams/ExamCreate.tsx]: avoid using React Fragment, use Mui Box instead

File: src/operations/exams/ExamCreate.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+              <AutocompleteInput
+                data-testId="awarded-course-input"
+                source="awarded_course_id"
+                choices={courseChoice}
+                label="Cours associé à un groupe"
+                optionText="courseName"
+                optionValue="id"
+                validate={required()}
+              />
+            )}{" "}
```
Comment (NyAndoMayah) [inline src/operations/exams/ExamCreate.tsx]: why did you put `{" "}`

File: src/operations/exams/ExamCreate.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            optionValue="id"
+            validate={required()}
+          />
+        )}
+
+        <TextInput
+          data-testId="title-input"
+          source="title"
+          label="Titre"
+          fullWidth
```
Comment (NyAndoMayah) [inline src/operations/exams/ExamCreate.tsx:108]: order your attributes

File: src/operations/exams/ExamCreate.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+import {DateTimeField} from "@/operations/common/components/fields";
+import {AutocompleteInput} from "@/ui/components/inputs";
+
+export const ExamCreate = () => {
+  const user = authProvider.getCachedWhoami();
+  const userRole = user.role;
+
+  const [selectedTeacherId, setSelectedTeacherId] = useState("");
+
+  const {data: awardedCourses = []} = useGetList("awarded-courses", {
```
Comment (NyAndoMayah) [inline src/operations/exams/ExamCreate.tsx:22]: can you explain me this

File: src/operations/exams/ExamCreate.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+  const {data: teacher = []} = useGetList("teachers");
+
+  const teacherAwardedChoice = useMemo(() => {
+    return teacherAwarded.map(({id, course, group}) => ({
+      id,
+      courseName: `${course.code} - ${group.ref}`,
+    }));
+  }, [teacherAwarded]);
+
+  const courseChoice = useMemo(() => {
```
Comment (NyAndoMayah) [inline src/operations/exams/ExamCreate.tsx]: typo

File: src/operations/exams/ExamCreate.tsx [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+  const handleTeacherChange = (value: string) => {
+    if (userRole !== "teacher") {
+      setSelectedTeacherId(value);
+    }
+  };
+
+  return (
+    <Create resource="exams" title="Création d'un examen">
+      <SimpleForm>
+        {userRole == "MANAGER" ? (
```
Comment (NyAndoMayah) [inline src/operations/exams/ExamCreate.tsx]: isManager in useRole

File: src/operations/exams/ExamList.tsx [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+          mainSearch={{label: "", source: ""}}
+        />
+        <ExamCard />
+      </List>
+    </Box>
+  );
+};
+
+export const ExamCard = () => {
+  const {data: exams = []} = useListContext();
```
Comment (NyAndoMayah) [inline src/operations/exams/ExamList.tsx:57]: go in line

File: src/operations/exams/components/SelectField.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  const [selectedValue, setSelectedValue] = useState<string | undefined>("");
+
+  const handleChange = (event: SelectChangeEvent<string>) => {
+    const value = event.target.value;
+    setSelectedValue(value);
+    onChange(value);
+  };
+
+  return (
+    <FormControl fullWidth variant="outlined" sx={{flex: 1}}>
```
Comment (NyAndoMayah) [inline src/operations/exams/components/SelectField.tsx]: order your attributes

File: src/providers/examProvider.tsx [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+const examsProvider: HaDataProviderType = {
+  getList: async (page, perPage, filter = {}, _meta) => {
+    const {
+      awarded_course_id,
+      title,
+      course_code,
+      group_ref,
+      examination_date_from,
+      examination_date_to,
+    } = filter;
```
Comment (NyAndoMayah) [inline src/providers/examProvider.tsx]: handle error here because what if one those attributes does not exist

**Review by NyAndoMayah — CHANGES_REQUESTED** [review]

File: src/operations/exams/ExamCreate.tsx (hunk 6 above)
Comment (Michellah) [reply src/operations/exams/ExamCreate.tsx:22]: it is to have the course associated with a teacher after he has selected the teacher

**Pull request merged**